lambda/src/main/scala/pricemigrationengine/handlers/AmendmentHandler.scala

```diff
@@ -29,9 +34,23 @@ object AmendmentHandler extends CohortHandler {
   ): ZIO[CohortTable with Zuora, Failure, AmendmentResult] =
     doAmendment(catalogue, item).foldZIO(
       failure = {
-        case _: CancelledSubscriptionFailure =>
+        case _: CancelledSubscriptionFailure => {
+          // `CancelledSubscriptionFailure` happens when the subscription was cancelled in Zuora
+          // in which case we simply update the processing state for this item in the database
+          // Although it was given to us as a failure of `doAmendement`, the only effect if the database update, it
+          // is not recorded as a failure of `amend` to allow the processing to continue.
           val result = CancelledAmendmentResult(item.subscriptionName)
           CohortTable.update(CohortItem.fromCancelledAmendmentResult(result)).as(result)
+        }
+        case _: StartDateNeedsToBeUpdatedFailure => {
+          // `StartDateNeedsToBeUpdatedFailure` is returned when we detect that the start date of an item is too close
+          // to the current date. This happens when, for instance, too long has passed between the creation of the
+          // cohort item and the actual Amendment. The side effect here is to update the item start date and to send
+          // it back to state `ReadyForEstimation`
+          val startDateNew = item.startDate.getOrElse(LocalDate.now()).plusDays(startDateShiftInCaseOfShortLeadTime)
```

Comment:
Best to avoid relative times as it's very difficult to test.  Use `zio.Clock` instead to get current time.

Reply:
Use `Clock.currentDateTime.map(_.toLocalDate)` instead of `LocalDate.now()`. Then we can use zio's [test clock](https://github.com/guardian/price-migration-engine/blob/0d252dc032fcfce1933a0ce7dd10da181f7b4ff5/lambda/src/test/scala/pricemigrationengine/handlers/EstimationHandlerTest.scala#L18) to set the time in tests.

Reply:
the startDate can't be calculated here. We need to let the estimation lambda generate it, so we just need to set it back to ReadyForEstimation and then change logic in the estimation handler so the earliestStartDate is 45 days forward (in this case Jess said we're only doing a 30 day lead time).